src/test/groovy/com/money_account_service/services/AccountServiceContractTest.groovy

```diff
@@ -0,0 +1,129 @@
+package com.money_account_service.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.money_account_service.dtos.response.AccountDetailsResponseDto
+import com.money_account_service.dtos.response.AuthorizeResponseDto
+import com.money_account_service.dtos.response.CreateAccountResponseDto
+import com.money_account_service.entities.AccountEntity
+import com.money_account_service.models.UserModel
+import org.spockframework.spring.SpringBean
+import com.money_account_service.repositories.AccountRepository
+import com.money_account_service.utility.UserServiceClient
+import okhttp3.Call
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.json.JSONObject
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import spock.lang.Specification
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class AccountServiceContractTest extends Specification {
+
+    @Autowired
+    private ObjectMapper objectMapper
+    @Autowired
+    private OkHttpClient okHttpClient
+    @SpringBean
+    private final AccountRepository accountRepository = Mock(AccountRepository)
+    @SpringBean
+    private final UserServiceClient userServiceClient = Mock(UserServiceClient)
+    private AuthorizationService authorizationService
+    private AccountService accountService
+
+    def setup() {
+        authorizationService = new AuthorizationService(userServiceClient)
```

Comment:
You can use `MockBean` and `SpyBean` features of SpringTest. Then it should replace the client inside the `AuthorizationService` I believe.